source/asynch_socket.c

```diff
@@ -153,13 +168,28 @@ static socket_error_t init()
     return SOCKET_ERROR_NONE;
 }
 
+void check_timeouts() {
+/**
+ * LwIP is not re-entrant for any APIs except pbuf and memp APIs.
+ * To correct for this, all packet processing should be deferred to event context.
+ * However, since sal-driver-lwip-k64f does all packet processing in IRQ context,
+ * we must use the same paradigm here until sal-driver-lwip-k64f is updated to defer
+ * packet processing to event context.
+ */
```

Comment:
Don't repeat yourself. (DRY)

It'd be nice to file a GitHub issue to keep track of removing this hack, and to refer to that issue anywhere the hack needs removing. It'd also be nice, for those who can't use GitHub or see git commit messages, to have the issue description (once) in this file.